core/src/main/scala/org/apache/spark/sql/row/JDBCMutableRelation.scala

```diff
@@ -90,9 +91,23 @@ case class JDBCMutableRelation(
   override def unhandledFilters(filters: Array[Filter]): Array[Filter] =
     filters.filter(ExternalStoreUtils.unhandledFilter)
 
-  protected final val connFactory: () => Connection =
-    JdbcUtils.createConnectionFactory(new JDBCOptions(connProperties.url, table,
-      connProperties.connProps.asScala.toMap))
+  protected final val connFactory: () => Connection = JdbcUtils
+      .createConnectionFactory(new JDBCOptions(connProperties.url +
+          connProperties.urlSecureSuffix,
+        table, connProperties.connProps.asScala.toMap))
+
+  protected final val sysConnFactory: () => Connection = {
+    val EMPTY_URL = ""
+    val user = sqlContext.conf.getConfString("snappydata.store.user", EMPTY_URL)
+    val password = sqlContext.conf.getConfString("snappydata.store.password", EMPTY_URL)
+    if (!user.equals(EMPTY_URL) && !password.equals(EMPTY_URL)) {
+      JdbcUtils.createConnectionFactory(new JDBCOptions(connProperties.url +
+          ";user=" + user +
+          ";password=" + password +
+          ";default-schema=" + SnappyStoreHiveCatalog.HIVE_METASTORE + ";",
+        table, connProperties.connProps.asScala.toMap))
+    } else connFactory
+  }
```

Comment:
Don't get this sysConnFactory changes in multiple places. Why is the "conn" itself not the correct authenticated one? In either case this code should be refactored into a separate utility method as mentioned.